mlflow/utils/search_utils.py

```diff
@@ -917,54 +920,41 @@ def paginate(cls, runs, page_token, max_results):
     VALID_SEARCH_KEYS_FOR_REGISTERED_MODELS = {"name"}
 
     @classmethod
-    def _check_valid_identifier_list(cls, value_token):
-        if len(value_token._groupable_tokens) == 0:
+    def _check_valid_identifier_list(cls, tup: tuple[Any, ...]) -> None:
+        """
+        Validate that `tup` is a non-empty tuple of strings.
+        """
+        if len(tup) == 0:
             raise MlflowException(
                 "While parsing a list in the query,"
                 " expected a non-empty list of string values, but got empty list",
                 error_code=INVALID_PARAMETER_VALUE,
             )
 
-        # Single element (e.g. `('x')`)
-        if (
-            len(value_token._groupable_tokens) == 1
-            and value_token._groupable_tokens[0].ttype is TokenType.String.Single
-        ):
-            return
-
-        # Multiple elements (e.g. `('x','y')`)
-        if not isinstance(value_token._groupable_tokens[0], IdentifierList):
-            raise MlflowException(
-                "While parsing a list in the query,"
-                " expected a non-empty list of string values, but got ill-formed list.",
-                error_code=INVALID_PARAMETER_VALUE,
-            )
-        elif not all(
-            token.ttype
-            in {*cls.STRING_VALUE_TYPES, *cls.DELIMITER_VALUE_TYPES, cls.WHITESPACE_VALUE_TYPE}
-            for token in value_token._groupable_tokens[0].tokens
-        ):
```

Comment:
We don't care what child tokens look like. We just care whether it represents a tuple of strings in python.